src/driver/platform_rpi.cpp

```diff
@@ -0,0 +1,140 @@
+
+/****************************************************************
+ * platform_rpi.cpp   
+ * HSCDTD008A Library Extention for RPI
+ * Tilman Glötzner
+ * Original Creation Date: 2022-09-07
+ *
+ * Distributed as-is; no warranty is given.
+ ***************************************************************/
+
+
+#include "platform.h"
+#include "hscdtd008a_driver.h"
+#include <stdio.h>
+#include <sys/ioctl.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <unistd.h>
+#include <string.h>
+#include <linux/i2c-dev.h>
+#define I2C_ADDR 0x0f
+#define device "/dev/i2c-1"
+#ifndef I2C_M_RD
+#include <linux/i2c.h>
+#endif
+
+int fd = -1;
+
+int8_t t_open(void)
+{
+    fd = open(device, O_RDWR); 
+    if (fd >= 0)
+    {
+       if (ioctl(fd, I2C_SLAVE, I2C_ADDR) < 0) {
+        printf("ioctl error: %s\n", strerror(errno));
+        return 1;
+       }
+    }   
+    return (fd<0);
+}
+
+int8_t t_read_register(uint8_t addr,
+                       uint8_t reg,
+                       uint8_t length,
+                       uint8_t *p_buffer)
+{
+    int8_t status;
+    uint8_t inbuf[32];
+    uint8_t outbuf[1];
+    struct i2c_msg msgs[2];
+    struct i2c_rdwr_ioctl_data msgset[1];
+
+    msgs[0].addr = addr;
+    msgs[0].flags = 0;
+    msgs[0].len = 1;
+    msgs[0].buf = outbuf;
+ 
+    outbuf[0] = reg;
+    
+    msgs[1].addr = addr;
+    msgs[1].flags = I2C_M_RD | I2C_M_NOSTART;
+    msgs[1].len = length;
+    msgs[1].buf = inbuf;
+
+    msgset[0].msgs = msgs;
+    msgset[0].nmsgs = 2;
```

Comment:
I am not familiar with I2C on the Raspberry Pi, could you please add some comments to explain what is happening here?